### ISSUE TITLE
ModeMCU ESP3266 - Can't Specify pin number as D0

### ISSUE THREAD

**Issue by Jimbobnz** [jasoncoon/esp8266-fastled-webserver#27]
```
....
#define DATA_PIN      D0    // for Huzzah: Pins w/o special function:  #4, #5, #12, #13, #14; // #16 does not work :(
#define LED_TYPE      WS2812
#define COLOR_ORDER   GRB
#define NUM_LEDS      8
......
```
```
Arduino: 1.8.1 (Windows 10), Board: "NodeMCU 0.9 (ESP-12 Module), 80 MHz, 115200, 4M (3M SPIFFS)"

In file included from C:\Users\TeamBowen\Documents\Arduino\esp8266-fastled-webserver\esp8266-fastled-webserver.ino:19:0:

C:\Users\TeamBowen\Documents\Arduino\libraries\FastLED-master/FastLED.h:17:21: note: #pragma message: FastLED version 3.001.005

 #    pragma message "FastLED version 3.001.005"

                     ^

In file included from C:\Users\TeamBowen\Documents\Arduino\libraries\FastLED-master/FastLED.h:65:0,

                 from C:\Users\TeamBowen\Documents\Arduino\esp8266-fastled-webserver\esp8266-fastled-webserver.ino:19:

C:\Users\TeamBowen\Documents\Arduino\libraries\FastLED-master/fastspi.h:110:23: note: #pragma message: No hardware SPI pins defined.  All SPI access will default to bitbanged output

 #      pragma message "No hardware SPI pins defined.  All SPI access will default to bitbanged output"

                       ^

In file included from C:\Users\TeamBowen\Documents\Arduino\libraries\FastLED-master/FastLED.h:48:0,

                 from C:\Users\TeamBowen\Documents\Arduino\esp8266-fastled-webserver\esp8266-fastled-webserver.ino:19:

C:\Users\TeamBowen\Documents\Arduino\libraries\FastLED-master/fastpin.h: In instantiation of 'class FastPin<16u>':

C:\Users\TeamBowen\Documents\Arduino\libraries\FastLED-master/platforms/esp/8266/clockless_esp8266.h:21:49:   required from 'class ClocklessController<16, 20, 50, 30, (EOrder)66u, 0, false, 5>'

C:\Users\TeamBowen\Documents\Arduino\libraries\FastLED-master/chipsets.h:468:7:   required from 'class WS2812Controller800Khz<16u, (EOrder)66u>'

C:\Users\TeamBowen\Documents\Arduino\libraries\FastLED-master/FastLED.h:100:52:   required from 'class WS2812<16u, (EOrder)66u>'

C:\Users\TeamBowen\Documents\Arduino\libraries\FastLED-master/FastLED.h:294:39:   required from 'static CLEDController& CFastLED::addLeds(CRGB*, int, int) [with CHIPSET = WS2812; unsigned char DATA_PIN = 16u; EOrder RGB_ORDER = (EOrder)66u]'

C:\Users\TeamBowen\Documents\Arduino\esp8266-fastled-webserver\esp8266-fastled-webserver.ino:105:66:   required from here

C:\Users\TeamBowen\Documents\Arduino\libraries\FastLED-master/fastpin.h:206:2: error: static assertion failed: Invalid pin specified

  static_assert(validpin(), "Invalid pin specified");

  ^

exit status 1
Error compiling for board NodeMCU 0.9 (ESP-12 Module).

This report would have more information with
"Show verbose output during compilation"
option enabled in File -> Preferences.

```

**Comment by jasoncoon** [comment]
Tried `#define DATA_PIN 0`?

**Comment by Jimbobnz** [comment]
I think found the issue accouding the . I'm needing to `#define FASTLED_ESP8266_NODEMCU_PIN_ORDER` before I `#include <FastLED.h>` in in the .ino file.

However this still not fixed the issue. In the end I got it working using the Adafruit Neopixel library instead.

Also I've got the feeling the the NodeMCU board is not fully supported platform due to the following GitHub Issue.
[](https://github.com/FastLED/FastLED/issues/429)
